tests/broadcast.rs

```diff
@@ -65,31 +62,17 @@ impl Adversary<Broadcast<NodeId>> for ProposeAdversary {
             return vec![];
         }
         self.has_sent = true;
-        let node_ids: BTreeSet<NodeId> = self
-            .adv_nodes
+        self.adv_nodes
             .iter()
-            .chain(self.good_nodes.iter())
-            .cloned()
-            .collect();
-        let id = match self.adv_nodes.iter().next() {
-            Some(id) => *id,
-            None => return vec![],
-        };
-
-        // FIXME: Take the correct, known keys from the network.
-        let netinfo = Arc::new(
-            NetworkInfo::generate_map(node_ids, &mut rand::thread_rng())
-                .expect("Failed to create `NetworkInfo` map")
-                .remove(&id)
-                .unwrap(),
-        );
-        let mut bc = Broadcast::new(netinfo, id).expect("broadcast instance");
-        // FIXME: Use the output.
-        let step = bc.handle_input(b"Fake news".to_vec()).expect("propose");
-        step.messages
-            .into_iter()
-            .map(|msg| MessageWithSender::new(id, msg))
-            .collect()
+            .flat_map(|(&id, netinfo)| {
+                Broadcast::new(netinfo.clone(), id)
+                    .expect("broadcast instance")
+                    .handle_input(b"Fake news".to_vec())
+                    .expect("propose")
+                    .messages
+                    .into_iter()
+                    .map(move |msg| MessageWithSender::new(id, msg))
```

Comment:
This still doesn't use `step.output`. Probably it's OK but don't we need to keep a note?

Reply:
Not really: I don't know why we had added that comment. After all, the adversary is just running fake nodes internally here, to create the fake messages to send.